fluid/PaddleCV/image_classification/eval_int8.py

```diff
@@ -0,0 +1,149 @@
+add_arg('image_shape',      str,  "3,224,224",         "Input image size")
+add_arg('with_mem_opt',     bool, True,                "Whether to use memory optimization or not.")
+add_arg('pretrained_model', str,  None,                "Whether to use pretrained model.")
+add_arg('skip_batch_num',   int,  5,                    "Skip batch num.")
+add_arg('use_transpiler',   bool, True,                 'Whether to use transpiler.')
+add_arg('use_fake_data',    bool, False,                'If set, use fake data instead of real data.')
+add_arg('iterations',	    int,  100,               	'Fake data iterations')
+add_arg('profiler',	    bool, False,                'If true, do profiling.')
+# yapf: enable
+
+model_list = [m for m in dir(models) if "__" not in m]
+
+
+def user_data_reader(data):
+    '''
+    Creates a data reader for user data.
+    '''
+
+    def data_reader():
+        while True:
+            for b in data:
+                yield b
+
+    return data_reader
+
+
+def eval(args):
+    # parameters from arguments
+    class_dim = args.class_dim
+    pretrained_model = args.pretrained_model
+    with_memory_optimization = args.with_mem_opt
+    skip_batch_num = args.skip_batch_num
+    if skip_batch_num >= args.iterations:
+       print("Please ensure the skip_batch_num less than iterations.")
+       sys.exit(0)
+    image_shape = [int(m) for m in args.image_shape.split(",")]
+
+
+    if with_memory_optimization:
+        fluid.memory_optimize(fluid.default_main_program())
+
+    place = fluid.CUDAPlace(0) if args.use_gpu else fluid.CPUPlace()
+    exe = fluid.Executor(place)
+
+    [infer_program, feed_dict,
+     fetch_targets] = fluid.io.load_inference_model(pretrained_model, exe)
+    
+    program = infer_program.clone()
+    if args.use_transpiler:
+        inference_transpiler_program = program.clone()
+        t = fluid.transpiler.InferenceTranspiler()
+        t.transpile(inference_transpiler_program, place)
+        program = inference_transpiler_program
+
+    fake_data = [(
+        np.random.rand(image_shape[0] * image_shape[1] * image_shape[2]).astype(np.float32),
+        np.random.randint(1, class_dim)) for _ in range(1)]
+
+    if args.use_fake_data:
+        val_reader = paddle.batch(
+            user_data_reader(fake_data), batch_size=args.batch_size)
+    else:
+        val_reader = paddle.batch(reader.val(), batch_size=args.batch_size)
+
+    test_info = [[], [], []]
+    cnt = 0
+    periods = []
+    for batch_id, data in enumerate(val_reader()):
+        if args.use_fake_data:
+            data = val_reader().next()
+        image = np.array(map(lambda x: x[0].reshape(image_shape), data)).astype(
+            "float32")
+        label = np.array(map(lambda x: x[1], data)).astype("int64")
+        label = label.reshape([-1, 1])
+
+        t1 = time.time()
+        loss, acc1, acc5 = exe.run(program, feed={feed_dict[0]: image, feed_dict[1]: label}, fetch_list=fetch_targets)
+
+        t2 = time.time()
```

Comment:
I'm a bit worried that this line being called everytime will affect performance.
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/executor.cc#L186

Reply:
We need to invoke the run several times independently:
1. Sampling on FP32: get tensor data, get feeder/fetch vars
2. Saving scale weights for INT8 program

Reply:
I mean the a problem in executor.cc might affect the time measured here. Not a problem of this PR
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/framework/executor.cc#L186

Reply:
got it. Actually, the measured time code is redundant here. We'll remove it in the new PR.